src/model/Params/NumberArrayParam.php

```diff
@@ -6,19 +6,22 @@ class NumberArrayParam extends BaseParam
 {
     protected $type = 'number_array';
 
-    private $options = false;
+    private $options;
 
-    public function __construct(string $key, string $label, string $help, bool $required = false, $default = null, $group = null, $options = null)
+    public function __construct(string $key, string $label, string $help, bool $required = false, $default = null, $group = null, ?array $options = null)
```

Comment:
Thanks for the PR!

Looking at the use, I'd say we could very well use this signature:

```
public function __construct(string $key, string $label, string $help, bool $required = false, $default = null, $group = null, array $options = [])
```

Since we don't do anything special with `null` value, we probably shouldn't allow it at all. What do you think?

Reply:
@riki137 Thanks for the `StringArrayParam` hint, I needed a refresh of these features.

For anyone stumbling upon this in the future, there **is** a difference between `$options` set and `$options` not set (`null`). If they're set, only provided set of values can be selected. If it's null, it's a free solo mode where people can enter any number/string they want.